js/ui/endSessionDialog.js

```diff
@@ -286,14 +286,34 @@ var EndSessionDialog = new Lang.Class({
         this._loginManager = LoginManager.getLoginManager();
         this._userManager = AccountsService.UserManager.get_default();
         this._user = this._userManager.get_user(GLib.get_user_name());
+        this._updatesPermission = false;
 
         this._pkOfflineProxy = new PkOfflineProxy(Gio.DBus.system,
                                                   'org.freedesktop.PackageKit',
                                                   '/org/freedesktop/PackageKit',
                                                   Lang.bind(this, function(proxy, error) {
-                                                      if (error)
+                                                      if (error) {
                                                           log(error.message);
+                                                          return;
+                                                      }
+
+                                                      // Creating a D-Bus proxy won't propagate SERVICE_UNKNOWN or NAME_HAS_NO_OWNER
+                                                      // errors if PackageKit is not available, but the GIO implementation will make
+                                                      // sure in that case that the proxy's g-name-owner is set to null, so check that.
+                                                      if (this._pkOfflineProxy.g_name_owner === null) {
+                                                          this._pkOfflineProxy = null;
+                                                          return;
+                                                      }
+
+                                                      // It only makes sense to check for this permission if PackageKit is available.
+                                                      try {
+                                                          this._updatesPermission =
+                                                              Polkit.Permission.new_sync("org.freedesktop.packagekit.trigger-offline-update", null, null);
+                                                      } catch(e) {
+                                                          log('No permission to trigger offline updates: %s'.format(e.toString()));
+                                                      }
```

Comment:
I personally would rather move this into a method of its own. It's too much code in a far indentation.

Reply:
I kind of agree, but at the same time I like having the code that initializes `this._updatesPermission` all in th e`_init()` function, so I'm not sure it's worth it.

Maybe comment the same thing in the upstream MR, and see what others (e.g. Florian) say about it?